.github/workflows/ci.yml

```diff
@@ -0,0 +1,577 @@
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-directories: ${{ env.CARGO_TARGET_DIR }}
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Build
+        run: |
+          ALL_CRATES="${PURELY_STD_CRATES} ${ALSO_WASM_CRATES}"
+          for crate in ${ALL_CRATES}; do
+            cargo build --verbose --all-features --release --manifest-path ./crates/${crate}/Cargo.toml;
+          done
+
+      - name: Build WASM
+        run: |
+          for crate in ${ALSO_WASM_CRATES}; do
+            cargo build --verbose --no-default-features --release --target wasm32-unknown-unknown \
+              --manifest-path ./crates/${crate}/Cargo.toml;
+          done
+
+  test:
+    runs-on: ubuntu-latest
+    needs: [check]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-directories: ${{ env.CARGO_TARGET_DIR }}
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Test
+        env:
+          # Fix for linking of `linkme` for `cargo test`: https://github.com/dtolnay/linkme/issues/49
+          RUSTFLAGS: -Clink-arg=-z -Clink-arg=nostart-stop-gc
+          # Since we run the tests with `--all-features` this implies the feature
+          # `ink-fuzz-tests` as well -- i.e. the fuzz tests are run.
+          # There's no way to disable a single feature while enabling all features
+          # at the same time, hence we use this workaround.
+          QUICKCHECK_TESTS:            0
+        run: |
+          cargo test --verbose --all-features --no-fail-fast --workspace
+          cargo test --verbose --all-features --no-fail-fast --workspace --doc
+
+  docs:
+    runs-on: ubuntu-latest
+    needs: [check, fmt, clippy, clippy-examples, dylint, spellcheck]
+    defaults:
+      run:
+        shell: bash
+    container:
+      image: paritytech/ci-unified:bullseye-1.73.0
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 100
+
+      - name: Cache
+        uses: Swatinem/rust-cache@v2
+        with:
+          cache-directories: ${{ env.CARGO_TARGET_DIR }}
+
+      - name: Rust Info
+        uses: ./.github/rust-info
+
+      - name: Create docs
+        env:
+          RUSTDOCFLAGS: -Dwarnings
+        run: |
+          # All crate docs currently need to be built separately. The reason
+          # is that `smart-bench-macro` is a dependency now in a number of places.
```

Comment:
This is no longer true, `smart-bench-macro` isn't a dependency any more. Does it mean we no longer need to build crate docs separately?

Reply:
We can remove it, but after such a change, 'ink_engine' and 'ink_allocator' have been added to the docs.
@ascjones please confirm that it is fine with you

Reply:
I think that is fine. AFAIU this task is just to check that the docs are built correctly? i.e. the resulting `crate-docs` are not used anywhere?

Reply:
@ascjones this job is used in 2 scenarios:
1. on pull-request, without any deployment
2. on push to master, with deployment to gh-pages branch